### ISSUE TITLE
Bug: BloodHoundCE web server unreachable after VM resume

### ISSUE THREAD

**Issue by h3retic** [SpecterOps/BloodHound#844]
## Description:

I am running BloodHoundCE via docker inside a VMWare Player virtual machine where I am doing hack the box prolabs etc. It works great initially, but if I stop the container for the night, suspend the VM, and then try to start the container again the next day I am unable to reach the web server at 127.0.0.1:8080. The only error in the output is:
 `bloodhound-1  | {"level":"error","time":"2024-09-08T18:44:17.070199108Z","message":"Invalid neo4j configuration supplied; returning default values"}`

### Are you intending to fix this bug?

No

## Component(s) Affected:

- UI
- API

## Steps to Reproduce:

1. Deploy bloodhoundce fresh via docker inside a VM
2. Login and change the password
3. Stop the docker container in the VM
4. Suspend the VM
5. Resume the VM
6. Restart the bloodhoundce docker container and observe error `Invalid neo4j configuration supplied; returning default values`
7. Attempt to access the web interface at 127.0.0.1:8080 and observe no response
8. Run `docker ps` and see that the bloodhound container is running and exposed at port 8080

## Expected Behavior:

Expect to be able to reach the web interface and login

## Actual Behavior:

Web interface is unresponsive:
```
└─$ curl 127.0.0.1:8080
curl: (56) Recv failure: Connection reset by peer
```
## Environment Information:

BloodHound: (I don't know how to determine the version, fresh build on 2024/09/07)
OS: `Linux kali 6.8.11-amd64 #1 SMP PREEMPT_DYNAMIC Kali 6.8.11-1kali2 (2024-05-30) x86_64 GNU/Linux`
Database: neo4j:4.4 / postgres:13.2
Docker: 27.0.3

## Additional Information:

I have tried restarting just the bloodhound container with no success. I have also tried leaving the containers running when I suspend the VM and sometimes that works and sometimes it is still unreachable. I also tried cleaning up the containers via `docker system prune -a`, making sure everything is deleted, and fetching latest, but the issue still occurs.

My only solution is to destroy and rebuild each time this happens. Below is a failed attempt to start bloodhoundce after resuming the virtual machine:
```                                                                                                                                                                          
[+] Running 3/0
 ✔ Container bloodhound-graph-db-1    Created                                                                                                                                                               0.0s 
 ✔ Container bloodhound-app-db-1      Created                                                                                                                                                               0.0s 
 ✔ Container bloodhound-bloodhound-1  Created                                                                                                                                                               0.0s 
Attaching to app-db-1, bloodhound-1, graph-db-1
app-db-1      | 
app-db-1      | PostgreSQL Database directory appears to contain a database; Skipping initialization
app-db-1      | 
app-db-1      | 2024-09-08 19:21:03.639 UTC [1] LOG:  starting PostgreSQL 13.2 (Debian 13.2-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
app-db-1      | 2024-09-08 19:21:03.639 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
app-db-1      | 2024-09-08 19:21:03.639 UTC [1] LOG:  listening on IPv6 address "::", port 5432
app-db-1      | 2024-09-08 19:21:03.640 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
app-db-1      | 2024-09-08 19:21:03.643 UTC [26] LOG:  database system was shut down at 2024-09-08 19:20:20 UTC
app-db-1      | 2024-09-08 19:21:03.647 UTC [1] LOG:  database system is ready to accept connections
graph-db-1    | Changed password for user 'neo4j'. IMPORTANT: this change will only take effect if performed before the database is started for the first time.
graph-db-1    | 2024-09-08 19:21:06.882+0000 INFO  Starting...
graph-db-1    | 2024-09-08 19:21:07.118+0000 INFO  This instance is ServerId{dedecdc1} (dedecdc1-5e1b-4e3e-8a05-77188af3ee92)
graph-db-1    | 2024-09-08 19:21:07.868+0000 INFO  ======== Neo4j 4.4.37 ========
graph-db-1    | 2024-09-08 19:21:08.483+0000 INFO  Performing postInitialization step for component 'security-users' with version 3 and status CURRENT
graph-db-1    | 2024-09-08 19:21:08.483+0000 INFO  Updating the initial password in component 'security-users'
graph-db-1    | 2024-09-08 19:21:09.466+0000 INFO  Bolt enabled on [0:0:0:0:0:0:0:0%0]:7687.
graph-db-1    | 2024-09-08 19:21:09.941+0000 INFO  Remote interface available at http://localhost:7474/
graph-db-1    | 2024-09-08 19:21:09.944+0000 INFO  id: 53E0935316D2CC6B355234AA3528DE8D63432741CA9B8CB7AB44B00CA25E1798
graph-db-1    | 2024-09-08 19:21:09.944+0000 INFO  name: system
graph-db-1    | 2024-09-08 19:21:09.944+0000 INFO  creationDate: 2024-09-08T19:12:53.924Z
graph-db-1    | 2024-09-08 19:21:09.944+0000 INFO  Started.
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:14.355186181Z","message":"Reading configuration found at /bloodhound.config.json"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:14.35560972Z","message":"Logging configured"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:14.367303715Z","message":"No database driver has been set for migration, using: neo4j"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:14.367356425Z","message":"Connecting to graph using Neo4j"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:14.370990442Z","message":"No new SQL migrations to run"}
bloodhound-1  | {"level":"error","time":"2024-09-08T19:21:15.483089159Z","message":"Invalid neo4j configuration supplied; returning default values"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.483146489Z","message":"Analysis requested by init"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.484277748Z","message":"Starting daemon API Daemon"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.484364518Z","message":"Starting daemon Tools API"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.484382808Z","message":"Starting daemon Data Pruning Daemon"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.484395168Z","message":"Starting daemon Data Pipe Daemon"}
bloodhound-1  | {"level":"info","time":"2024-09-08T19:21:15.484435718Z","message":"Server started successfully"}
```
And the `docker ps` output:
```
└─$ docker ps                                                                                                                                                                                                    
CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS                   PORTS                                                          NAMES
21fee4fd34c4   specterops/bloodhound:latest   "/bloodhound -config…"   10 minutes ago   Up 2 minutes             127.0.0.1:8080->8080/tcp                                       bloodhound-bloodhound-1
c82aab2c82fb   postgres:13.2                  "docker-entrypoint.s…"   10 minutes ago   Up 2 minutes (healthy)   5432/tcp                                                       bloodhound-app-db-1
4251fa0292fe   neo4j:4.4                      "tini -g -- /startup…"   10 minutes ago   Up 2 minutes (healthy)   127.0.0.1:7474->7474/tcp, 7473/tcp, 127.0.0.1:7687->7687/tcp   bloodhound-graph-db-1
```
And curl demonstrating the unresponsive web interface:
```
└─$ curl 127.0.0.1:8080
curl: (56) Recv failure: Connection reset by peer
```
## Contributor Checklist:

- [X] I have searched the issue tracker to ensure this bug hasn't been reported before or is not already being addressed.
- [X] I have provided clear steps to reproduce the issue.
- [X] I have included relevant environment information details.

**Comment by h3retic** [comment]
Netstat shows the docker port:
`tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      624115/docker-proxy`

Attempting to /bin/bash in the container fails (or /bin/sh):
```
└─$ docker ps
CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS                   PORTS                                                          NAMES
21fee4fd34c4   specterops/bloodhound:latest   "/bloodhound -config…"   47 minutes ago   Up 5 minutes             127.0.0.1:8080->8080/tcp                                       bloodhound-bloodhound-1
c82aab2c82fb   postgres:13.2                  "docker-entrypoint.s…"   47 minutes ago   Up 6 minutes (healthy)   5432/tcp                                                       bloodhound-app-db-1
4251fa0292fe   neo4j:4.4                      "tini -g -- /startup…"   47 minutes ago   Up 6 minutes (healthy)   127.0.0.1:7474->7474/tcp, 7473/tcp, 127.0.0.1:7687->7687/tcp   bloodhound-graph-db-1

└─$ sudo docker exec -it 21fee4fd34c4 /bin/bash
OCI runtime exec failed: exec failed: unable to start container process: exec: "/bin/bash": stat /bin/bash: no such file or directory: unknown
```

I tried restarting network manager and also instructing it to unmanage docker interfaces with the following config:
```
[keyfile]
unmanaged-devices=interface-name:docker*;interface-name:veth*;interface-name:br-*;interface-name:vmnet*;interface-name:vboxnet*
```

**Comment by h3retic** [comment]
Good news, looks like modifying network manager might have actually fixed it, I had to suspend and resume first to see the results (just restarting network manager didn't do it). I found the fix on stack overflow [cant-connect-to-docker-after-resuming-vm](https://stackoverflow.com/questions/57874055/cant-connect-to-docker-after-resuming-vm).

Edit the file: `/etc/NetworkManager/conf.d/10-unmanage-docker-interfaces.conf` and add the below lines:
```
[keyfile]
unmanaged-devices=interface-name:docker*;interface-name:veth*;interface-name:br-*;interface-name:vmnet*;interface-name:vboxnet*
```

This will mark the interfaces as unmanaged so Network Manager won't mess with them, apparently this fixes it! Sorry for creating this issue that isn't caused by bloodhound, but hopefully it can be of help to other people that may run into this problem.
